### PR TITLE
Integrate @hashicorp/react-search into layout

### PR DESCRIPTION
**Pull request by pruett**
This PR updates the UI to expose search capability via our `@hashicorp/react-search` react component and our Algolia-powered index.

Screen Shot 2020-09-01 at 9 12 22 AM

### CODE REVIEW

**Review by samsalisbury — COMMENTED** [review]
Hi, I'm not sure if there was a specific part of this you wanted me to review? Usually I'd be concerned with reviewing changes to the build system (`packakges-oss.yml`) or CI config (`.circleci/`) but I don't see any changes to those paths. Happy to review again if there's something specific you'd like me to look at.

**Comment by pruett** [comment]
Thanks @samsalisbury . As mentioned [here](https://github.com/hashicorp/vault/pull/9867), I'll unassign you from this PR as the small part I wanted to alert you of (the `.env` file change) is largely a frontend concern. Thanks 🙏

**Review by kalafut — APPROVED** [review]

**Review by calvn — APPROVED** [review]
Looks great, excited for this! Is this functional on local builds as well?

**Comment by pruett** [comment]
@calvn Indeed it is functional on local builds!

**Comment by pruett** [comment]
@kylemac @jescalan Now includes extracted Algolia indexing logic called over at `website/scripts/index_search_content.js`

File: website/components/search-bar/style.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.g-search {
+  width: calc(100% - 2rem);
+  max-width: 600px;
+
+  & .c-hits .ais-Highlight-highlighted {
+    /* Add custom background color */
```
Comment (jescalan) [inline website/components/search-bar/style.css:6]: This should be completed or removed!

**Review by jescalan — APPROVED** [review]

**Pull request merged**